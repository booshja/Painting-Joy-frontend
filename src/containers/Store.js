// dependencies
import React, { useContext, useEffect, useState } from "react";
import styled from "styled-components";
import axios from "axios";
// components
import { LoadingSpinner, PageTitle, StoreItem } from "../components";
// context
import ItemsContext from "../context/ItemsContext";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledStore = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 45vh;

    ${breakpoints("min-height", "vh", [{ 414: 49 }, { 768: 52 }])}
`;

const StyledItemsContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    width: 95%;

    ${breakpoints("justify-content", "", [{ 768: "center" }])}
`;

const StyledP = styled.p`
    font-family: "News Cycle";
    font-size: 1.4rem;
`;

/** TEMPORARY 'COMING SOON' BANNER */
const StyledBanner = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;

    p {
        font-family: "News Cycle", sans-serif;
        font-size: 2rem;
    }
`;
/** END TEMPORARY 'COMING SOON' BANNER */

const Store = () => {
    // set up state
    // const [loading, setLoading] = useState(true);

    // set up context
    const { items, setItems } = useContext(ItemsContext);

    // useEffect(() => {
    //     // get cancel token for axios
    //     const source = axios.CancelToken.source();

    //     const getItems = async () => {
    //         // on mount, get items data from store
    //         const itemRes = await axios.get(
    //             process.env.REACT_APP_BACKEND_URL + "items/",
    //             { cancelToken: source.token }
    //         );
    //         setItems(itemRes.data.items);
    //         setLoading(false);
    //     };
    //     getItems();

    //     return function cleanup() {
    //         // on unmount cancel any open axios requests
    //         source.cancel();
    //     };
    // }, []);

    // if (loading)
    //     return (
    //         // if loading, return loading spinner
    //         <StyledStore>
    //             <LoadingSpinner />
    //         </StyledStore>
    //     );

    return (
        <StyledStore>
            <PageTitle>Store</PageTitle>
            {/* {items ? (
                <StyledItemsContainer>
                    {items.map((item) => (
                        <StoreItem item={item} key={item.id} />
                    ))}
                </StyledItemsContainer>
            ) : (
                <StyledP>Nothing Currently For Sale!</StyledP>
            )} */}
            <StyledBanner>
                <p>Store coming soon!</p>
            </StyledBanner>
        </StyledStore>
    );
};

export default Store;
