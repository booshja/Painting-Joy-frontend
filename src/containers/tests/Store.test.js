import { render } from "@testing-library/react";
import Store from "../Store";

/** Smoke Test */
it("renders without crashing", () => {
    render(<Store />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<Store />);
    expect(asFragment()).toMatchSnapshot();
});
