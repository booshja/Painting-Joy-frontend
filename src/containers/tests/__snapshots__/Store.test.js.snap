// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  color: #207070;
  font-size: 5rem;
  margin-top: 2rem;
}

.c2 {
  font-size: 1.4rem;
  color: #207070;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-top: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 45vh;
}

@media screen and (min-width:414px) {
  .c0 {
    min-height: 49vh;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    min-height: 52vh;
  }
}

<div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-palette fa-w-16 fa-spin c1"
      data-icon="palette"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M204.3 5C104.9 24.4 24.8 104.3 5.2 203.4c-37 187 131.7 326.4 258.8 306.7 41.2-6.4 61.4-54.6 42.5-91.7-23.1-45.4 9.9-98.4 60.9-98.4h79.7c35.8 0 64.8-29.6 64.9-65.3C511.5 97.1 368.1-26.9 204.3 5zM96 320c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm32-128c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128-64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
        fill="currentColor"
      />
    </svg>
    <p
      class="c2"
    >
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
